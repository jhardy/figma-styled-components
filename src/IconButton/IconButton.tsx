import styled, { css } from 'styled-components'
import { OptionButton, OptionButtonContent } from '../OptionButton'


const getHoverStyles = (on: boolean, disabled: boolean) => {
  if (disabled) {
    return css`
      background: transparent;
    `
  } else {
    return css`
      background: ${on ? '#18A0FB' : 'rgba(0, 0, 0, 0.1)'};
    `
  }
}

const IconButtonFactory = styled(OptionButton)`
   background: ${(props) => props.on ? '#18A0FB' : 'transparent'};
   color: ${(props) => props.on ? '#ffffff' : '#000000'};

    &:hover {
      ${(props) => getHoverStyles(props.on || false, props.disabled || false)}
      box-shadow: none;
    }

    &:active {
      background: ${(props) => props.disabled ? 'transparent' : '#18A0FB'};
      color: ${(props) => props.disabled ? '#000' : '#fff'};

      ${OptionButtonContent} {
        background: transparent;
      }
    }

    &:focus {
      box-shadow: ${(props) => props.on ? 'inset 0 0 0 2px rgba(0, 0, 0, 0.3)' : 'inset 0 0 0 2px #18A0FB'};
    }

   ${OptionButtonContent} {
     background: transparent;
   }
`
export const IconButton = styled(IconButtonFactory)``
