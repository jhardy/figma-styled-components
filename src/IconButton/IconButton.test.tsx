
import 'jest-styled-components'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { IconButton } from './IconButton'


const testIcon = (
  <svg width='16' height='16' viewBox='0 0 16 16' fill='none' xmlns='http://www.w3.org/2000/svg'>
<path fillRule='evenodd' clip-rule='evenodd' d='M11 4.5V6H12V4.5C12 2.567 10.433 1 8.5 1C6.567 1 5 2.567 5 4.5V6H6V4.5C6 3.11929 7.11929 2 8.5 2C9.88071 2 11 3.11929 11 4.5ZM11 10H12V11.5C12 13.433 10.433 15 8.5 15C6.567 15 5 13.433 5 11.5V10H6V11.5C6 12.8807 7.11929 14 8.5 14C9.88071 14 11 12.8807 11 11.5V10ZM8 5V11H9V5H8Z' fill='black' fill-opacity='0.8'/>
</svg>

)

test('A default Option Button', () => {
  const tree = renderer.create(<IconButton icon={testIcon} />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('A on Option Button', () => {
  const tree = renderer.create(<IconButton toggled={true} icon={testIcon} />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('A dsiabled Option Button', () => {
  const tree = renderer.create(<IconButton disabled={true} icon={testIcon} />).toJSON()
  expect(tree).toMatchSnapshot()
})
