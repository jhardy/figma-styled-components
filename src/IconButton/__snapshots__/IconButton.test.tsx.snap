// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A default Option Button 1`] = `
.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 2px;
  background: transparent;
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 24px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 1;
  outline: none;
  pointer-events: none;
  text-align: center;
  min-width: 24px;
  width: 24px;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: transparent;
  border: none;
  border-radius: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 32px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 32px;
  outline: none;
  color: #000;
}

.c1:hover {
  box-shadow: inset 0 0 0 1px rgba(0,0,0,0.1);
}

.c1:active .c2 {
  background: rgba(0,0,0,0.1);
}

.c1:focus {
  box-shadow: inset 0 0 0 2px #18A0FB;
}

.c0 {
  background: transparent;
  color: #000000;
}

.c0:hover {
  background: rgba(0,0,0,0.1);
  box-shadow: none;
}

.c0:active {
  background: #18A0FB;
  color: #fff;
}

.c0:active .c2 {
  background: transparent;
}

.c0:focus {
  box-shadow: inset 0 0 0 2px #18A0FB;
}

.c0 .c2 {
  background: transparent;
}

<button
  className="c0 c1"
  disabled={false}
>
  <div
    className="c2 c3"
    disabled={false}
  >
    <svg
      fill="none"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M11 4.5V6H12V4.5C12 2.567 10.433 1 8.5 1C6.567 1 5 2.567 5 4.5V6H6V4.5C6 3.11929 7.11929 2 8.5 2C9.88071 2 11 3.11929 11 4.5ZM11 10H12V11.5C12 13.433 10.433 15 8.5 15C6.567 15 5 13.433 5 11.5V10H6V11.5C6 12.8807 7.11929 14 8.5 14C9.88071 14 11 12.8807 11 11.5V10ZM8 5V11H9V5H8Z"
        fill="black"
        fill-opacity="0.8"
        fillRule="evenodd"
      />
    </svg>
  </div>
</button>
`;

exports[`A dsiabled Option Button 1`] = `
.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 2px;
  background: transparent;
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 24px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 0.3;
  outline: none;
  pointer-events: none;
  text-align: center;
  min-width: 24px;
  width: 24px;
}

.c4:active .c2 {
  background: rgba(0,0,0,0.1);
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: transparent;
  border: none;
  border-radius: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 32px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 32px;
  outline: none;
  color: #000;
}

.c1:hover {
  box-shadow: inset 0 0 0 1px rgba(0,0,0,0.1);
}

.c1:active .c2 {
  background: transparent;
}

.c1:focus {
  box-shadow: inset 0 0 0 2px #18A0FB;
}

.c5:active .c2 {
  background: transparent;
}

.c5 .c2 {
  background: transparent;
}

.c6:active .c2 {
  background: transparent;
}

.c6 .c2 {
  background: transparent;
}

.c0 {
  background: transparent;
  color: #000000;
}

.c0:hover {
  background: transparent;
  box-shadow: none;
}

.c0:active {
  background: transparent;
  color: #000;
}

.c0:active .c2 {
  background: transparent;
}

.c0:focus {
  box-shadow: inset 0 0 0 2px #18A0FB;
}

.c0 .c2 {
  background: transparent;
}

<button
  className="c0 c1"
  disabled={true}
>
  <div
    className="c2 c3"
    disabled={true}
  >
    <svg
      fill="none"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M11 4.5V6H12V4.5C12 2.567 10.433 1 8.5 1C6.567 1 5 2.567 5 4.5V6H6V4.5C6 3.11929 7.11929 2 8.5 2C9.88071 2 11 3.11929 11 4.5ZM11 10H12V11.5C12 13.433 10.433 15 8.5 15C6.567 15 5 13.433 5 11.5V10H6V11.5C6 12.8807 7.11929 14 8.5 14C9.88071 14 11 12.8807 11 11.5V10ZM8 5V11H9V5H8Z"
        fill="black"
        fill-opacity="0.8"
        fillRule="evenodd"
      />
    </svg>
  </div>
</button>
`;

exports[`A on Option Button 1`] = `
.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 2px;
  background: rgba(0,0,0,0.1);
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 24px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 1;
  outline: none;
  pointer-events: none;
  text-align: center;
  min-width: 24px;
  width: 24px;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: transparent;
  border: none;
  border-radius: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 32px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 32px;
  outline: none;
  color: #000;
}

.c1:hover {
  box-shadow: inset 0 0 0 1px rgba(0,0,0,0.1);
}

.c1:active .c2 {
  background: rgba(0,0,0,0.1);
}

.c1:focus {
  box-shadow: inset 0 0 0 2px #18A0FB;
}

.c4:active .c2 {
  background: transparent;
}

.c4 .c2 {
  background: transparent;
}

.c0 {
  background: #18A0FB;
  color: #ffffff;
}

.c0:hover {
  background: #18A0FB;
  box-shadow: none;
}

.c0:active {
  background: #18A0FB;
  color: #fff;
}

.c0:active .c2 {
  background: transparent;
}

.c0:focus {
  box-shadow: inset 0 0 0 2px rgba(0,0,0,0.3);
}

.c0 .c2 {
  background: transparent;
}

<button
  className="c0 c1"
  disabled={false}
>
  <div
    className="c2 c3"
    disabled={false}
  >
    <svg
      fill="none"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M11 4.5V6H12V4.5C12 2.567 10.433 1 8.5 1C6.567 1 5 2.567 5 4.5V6H6V4.5C6 3.11929 7.11929 2 8.5 2C9.88071 2 11 3.11929 11 4.5ZM11 10H12V11.5C12 13.433 10.433 15 8.5 15C6.567 15 5 13.433 5 11.5V10H6V11.5C6 12.8807 7.11929 14 8.5 14C9.88071 14 11 12.8807 11 11.5V10ZM8 5V11H9V5H8Z"
        fill="black"
        fill-opacity="0.8"
        fillRule="evenodd"
      />
    </svg>
  </div>
</button>
`;
